[Bug] Default keymaps for both of my keyboards fail

### Describe the Bug
Getting this error when compiling the default keymap for my keyboards

```
❯ qmk compile -kb kbdfans/kbd67/mkii_soldered -km default
Ψ Compiling keymap with make -r -R -f builddefs/build_keyboard.mk -s KEYBOARD=kbdfans/kbd67/mkii_soldered KEYMAP=default KEYBOARD_FILESAFE=kbdfans_kbd67_mkii_soldered TARGET=kbdfans_kbd67_mkii_soldered_default VERBOSE=false COLOR=true SILENT=false QMK_BIN="qmk"


Generating: .build/obj_kbdfans_kbd67_mkii_soldered_default/src/default_keyboard.c                   [OK]
avr-gcc (GCC) 15.1.0
Copyright (C) 2025 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Generating: .build/obj_kbdfans_kbd67_mkii_soldered_default/src/info_config.h                        [OK]
Generating: .build/obj_kbdfans_kbd67_mkii_soldered_default/src/default_keyboard.h                   [OK]
Compiling: .build/obj_kbdfans_kbd67_mkii_soldered_default/src/default_keyboard.c                    [OK]
Compiling: quantum/keymap_introspection.c                                                           [OK]
Compiling: quantum/quantum.c                                                                        [OK]
Compiling: quantum/bitwise.c                                                                        [OK]
Compiling: quantum/led.c                                                                            [OK]
Compiling: quantum/action.c                                                                         [OK]
Compiling: quantum/action_layer.c                                                                   [OK]
Compiling: quantum/action_tapping.c                                                                 [OK]
Compiling: quantum/action_util.c                                                                    [OK]
Compiling: quantum/eeconfig.c                                                                       [OK]
Compiling: quantum/keyboard.c                                                                       [OK]
Compiling: quantum/keymap_common.c                                                                  [OK]
Compiling: quantum/keycode_config.c                                                                 [OK]
Compiling: quantum/sync_timer.c                                                                     [OK]
Compiling: quantum/logging/debug.c                                                                  [OK]
Compiling: quantum/logging/sendchar.c                                                               [OK]
Compiling: quantum/process_keycode/process_default_layer.c                                          [OK]
Compiling: quantum/process_keycode/process_oneshot.c                                                [OK]
Compiling: quantum/process_keycode/process_quantum.c                                                [OK]
Compiling: quantum/nvm/eeprom/nvm_eeconfig.c                                                        [OK]
Compiling: quantum/matrix_common.c                                                                  [OK]
Compiling: quantum/matrix.c                                                                         [OK]
Compiling: quantum/debounce/sym_defer_g.c                                                           [OK]
Compiling: quantum/main.c                                                                           [OK]
Assembling: platforms/avr/xprintf.S                                                                 [OK]
Compiling: platforms/avr/printf.c                                                                   [OK]
Compiling: quantum/backlight/backlight.c                                                            [OK]
Compiling: quantum/process_keycode/process_backlight.c                                              [OK]
Compiling: quantum/backlight/backlight_driver_common.c                                              [OK]
Compiling: platforms/avr/drivers/backlight_pwm.c                                                    [OK]
Compiling: quantum/bootmagic/bootmagic.c                                                            [OK]
Compiling: quantum/command.c                                                                        [OK]
Compiling: quantum/process_keycode/process_grave_esc.c                                              [OK]
Compiling: quantum/process_keycode/process_magic.c                                                  [OK]
Compiling: quantum/mousekey.c                                                                       [OK]
Compiling: quantum/send_string/send_string.c                                                       quantum/send_string/send_string.c: In function 'tap_random_base64':
quantum/send_string/send_string.c:286:20: error: 'TCNT0' undeclared (first use in this function)
  286 |     uint8_t key = (TCNT0 + TCNT1 + TCNT3 + TCNT4) % 64;
      |                    ^~~~~
quantum/send_string/send_string.c:286:20: note: each undeclared identifier is reported only once for each function it appears in
quantum/send_string/send_string.c:286:28: error: 'TCNT1' undeclared (first use in this function)
  286 |     uint8_t key = (TCNT0 + TCNT1 + TCNT3 + TCNT4) % 64;
      |                            ^~~~~
quantum/send_string/send_string.c:286:36: error: 'TCNT3' undeclared (first use in this function)
  286 |     uint8_t key = (TCNT0 + TCNT1 + TCNT3 + TCNT4) % 64;
      |                                    ^~~~~
quantum/send_string/send_string.c:286:44: error: 'TCNT4' undeclared (first use in this function)
  286 |     uint8_t key = (TCNT0 + TCNT1 + TCNT3 + TCNT4) % 64;
      |                                            ^~~~~
 [ERRORS]
 |
 |
 |
make: *** [builddefs/common_rules.mk:362: .build/obj_kbdfans_kbd67_mkii_soldered_default/quantum/send_string/send_string.o] Error 1
```
same for 
```
❯ qmk compile -kb kbdfans/kbd67/rev2 -km default
```

### Keyboard Used

kbdfans/kbd67/rev2
and
kbdfans/kbd67/mkii_soldered

### Operating System

Arch Linux

### qmk doctor Output

```text
❯ qmk doctor
Ψ QMK Doctor is checking your environment.
Ψ Python version: 3.13.11
Ψ CLI version: 1.1.8
Ψ QMK home: /home/roland/development/qmk_firmware
Ψ Detected Linux (Arch Linux).
<class 'FileNotFoundError'>
☒ [Errno 2] No such file or directory: '/etc/udev/rules.d/45-ddcutil-i2c.rules'
Traceback (most recent call last):
  File "/usr/lib/python3.13/site-packages/milc/milc.py", line 606, in __call__
    return self.__call__()
           ~~~~~~~~~~~~~^^
  File "/usr/lib/python3.13/site-packages/milc/milc.py", line 611, in __call__
    return self._subcommand(self)
           ~~~~~~~~~~~~~~~~^^^^^^
  File "/home/roland/development/qmk_firmware/lib/python/qmk/cli/doctor/main.py", line 184, in doctor
    status = os_status = os_tests()
                         ~~~~~~~~^^
  File "/home/roland/development/qmk_firmware/lib/python/qmk/cli/doctor/main.py", line 82, in os_tests
    return os_test_linux()
  File "/home/roland/development/qmk_firmware/lib/python/qmk/cli/doctor/linux.py", line 151, in os_test_linux
    rc = check_udev_rules()
  File "/home/roland/development/qmk_firmware/lib/python/qmk/cli/doctor/linux.py", line 86, in check_udev_rules
    for line in rule_file.read_text(encoding='utf-8').split('\n'):
                ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.13/pathlib/_local.py", line 546, in read_text
    return PathBase.read_text(self, encoding, errors, newline)
           ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.13/pathlib/_abc.py", line 632, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors, newline=newline) as f:
         ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.13/pathlib/_local.py", line 537, in open
    return io.open(self, mode, buffering, encoding, errors, newline)
           ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/etc/udev/rules.d/45-ddcutil-i2c.rules'

qmk_firmware on  um [$?] via  pyenv system ☁
❯ yay -Ql ddcutil|grep i2c
ddcutil /usr/lib/udev/rules.d/60-ddcutil-i2c.rules
ddcutil /usr/share/ddcutil/data/60-ddcutil-i2c.rules
ddcutil /usr/share/ddcutil/data/90-nvidia-i2c.conf
ddcutil /usr/share/ddcutil/data/nvidia-i2c.conf

qmk_firmware on  um [$?] via  pyenv system ☁
❯ yay -Ql ddcutil|grep ddcutil-i2c
ddcutil /usr/lib/udev/rules.d/60-ddcutil-i2c.rules
ddcutil /usr/share/ddcutil/data/60-ddcutil-i2c.rules

qmk_firmware on  um [$?] via  pyenv system ☁
❯ cat /usr/lib/udev/rules.d/60-ddcutil-i2c.rules
SUBSYSTEM=="i2c-dev", KERNEL=="i2c-[0-9]*", ATTRS{class}=="0x03*", TAG+="uaccess"
SUBSYSTEM=="dri",KERNEL=="card[0-9]*", TAG+="uaccess"
```

--- a/quantum/send_string/send_string.c
+++ b/quantum/send_string/send_string.c
@@ -281,6 +281,10 @@ void send_nibble(uint8_t number) {
     }
 }
 
+#if defined(__AVR_ATmega32U4__)
+#    include <avr/io.h>
+#endif
+
 void tap_random_base64(void) {
 #if defined(__AVR_ATmega32U4__)
     uint8_t key = (TCNT0 + TCNT1 + TCNT3 + TCNT4) % 64;

